wdl-ast/src/v1/expr.rs

```diff
@@ -7837,4 +7839,38 @@ task test {
             _ => panic!("expected text part"),
         }
     }
+
+    #[test]
+    fn strip_whitespace_with_content_on_first_line() {
+        let (document, diagnostics) = Document::parse(
+            r#"
+version 1.2
+
+task test {
+    String hw = <<<    hello world
+    my name is Jeff.
+    >>>
+}"#,
+        );
+
+        assert!(diagnostics.is_empty());
+        let ast = document.ast();
+        let ast = ast.as_v1().expect("should be a V1 AST");
+
+        let tasks: Vec<_> = ast.tasks().collect();
+        assert_eq!(tasks.len(), 1);
+
+        let decls: Vec<_> = tasks[0].declarations().collect();
+        assert_eq!(decls.len(), 1);
+
+        let expr = decls[0].expr().unwrap_literal().unwrap_string();
+        let stripped = expr.strip_whitespace().unwrap();
+        assert_eq!(stripped.len(), 1);
+        match &stripped[0] {
+            StrippedStringPart::Text(text) => {
+                assert_eq!(text.as_str(), "hello world\n    my name is Jeff.")
```

Comment:
@peterhuene is this correct, or should it be `"hello world\nmy name is Jeff."`?

Reply:
From my reading of the spec, I _think_ this is the expected output.

According to the spec the order of operations is:

1. Remove all line continuations and subsequent white space.
2. Remove all whitespace following the opening <<<, up to and including a newline (if any).
3. Remove all whitespace preceeding the closing >>>, up to and including a newline (if any).
4. Use all remaining non-blank lines to determine the common leading whitespace. 
5. Strip leading whitespace.

I think we're now faithfully following these steps; we remove line continuations first, then the `<<<` and `>>>` whitespace, then we calculate the minimum leading whitespace from every non-blank line, then we strip the leading whitespace.

In this case, as we remove the whitespace on the first line in step 2, we determine the minimum leading whitespace to be 0 because `hello world` is the first non-blank line; as a result, the leading whitespace remains.